libethash-cl/kernels/cl/ethash.cl

```diff
@@ -236,10 +236,20 @@ do { \
 
 #endif
 
+typedef struct {
+    uint gid;
+    uint mix[8];
+    uint pad[7]; // pad to 16  words for easy indexing
+} SearchResult;
+
+typedef struct {
+    SearchResult rslt[MAX_OUTPUTS];
+    uint count;
+} SearchResults;
 
 __attribute__((reqd_work_group_size(WORKSIZE, 1, 1)))
 __kernel void search(
-    __global volatile uint* restrict g_output,
+    __global volatile SearchResults* restrict g_output,
```

Comment:
Why `volatile` is needed here?

Reply:
Not sure why? Was there in the original. Probably not necessary...